tests/integration-tests/tests/createami/test_createami.py

```diff
@@ -116,8 +103,38 @@ def test_createami_post_install(
         + post_install_args
     )
 
-    assert_that(
-        any("No post install script".lower() in s.lower() for s in pcluster_createami_result.stdout.split("\n"))
-    ).is_false()
+    stdout_lower = pcluster_createami_result.stdout.lower()
+    assert_that(stdout_lower).does_not_contain("no post install script")
+    assert_that(stdout_lower).does_not_contain("no custom ami created")
+
 
-    assert_that(pcluster_createami_result.stdout).does_not_contain("No custom AMI created")
+@pytest.mark.dimensions("eu-central-1", "c5.xlarge", "alinux", "sge")
+def test_createami_wrong_os(
+    region, os, instance, test_datadir, request, pcluster_config_reader, vpc_stack, architecture, scheduler
+):
+    """Test error message when os provide is different from the os of custom AMI"""
+    cluster_config = pcluster_config_reader()
+
+    # alinux is specified in the config file but an AMI of alinux2 is provided
+    wrong_os = "alinux2"
```

Comment:
Similar to comment above, let's do a quick assertion here to make sure wrong_os != os